crates/net/src/eventsource/mod.rs

```diff
@@ -0,0 +1,38 @@
+//! Wrapper around the `EventSource` API
+//!
+//! This API is provided in the following flavors:
+//! - [Futures API][futures]
+
+pub mod futures;
+
+use std::fmt;
+
+/// The state of the EventSource.
+///
+/// See [`EventSource.readyState` on MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventSource/readyState)
+/// to learn more.
+#[derive(Copy, Clone, Debug)]
+pub enum State {
+    /// The connection has not yet been established.
+    Connecting,
+    /// The EventSource connection is established and communication is possible.
+    Open,
+    /// The connection has been closed or could not be opened.
+    Closed,
+}
+
+/// Error returned by the EventSource
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum EventSourceError {
```

Comment:
Why not use `thiserror` here?

Reply:
I'm unfamiliar with `thiserror`. However, am I not doing the same thing here as per `WebSocket` by way of declaring the error type?

Reply:
I really like `thiserror`, but I agree with @huntc that this crate (at least in websocket/mod.rs) is not using `thiserror`. Is this something we should consider changing consistently in another PR, @hamza1311? I'd be glad to provide one.

Reply:
@hseeberger if you want to PR converting this to `thiserror`, that would be great.

It's fine as is for this PR